eth/downloader/downloader.go

```diff
@@ -1033,8 +1035,15 @@ func (d *Downloader) fetchHeaders(p *peerConnection, from uint64, pivot uint64)
 				from += uint64(len(headers))
 				getHeaders(from)
 			} else {
-				// No headers delivered, or all of them being delayed, sleep a bit and retry
+				// No headers delivered
+				if !skeleton && ignoredHeaders == 0 && from < advertisedHead {
+					// This peer told is about a high number, but is not
+					// delivering
+					p.log.Trace("peer did not deliver", "requested", from, "head", advertisedHead)
```

Comment:
Pls capitalise `Peer` (log), also perhaps lets bump this too to Warn?

Reply:
Perhaps `"Peer did not deliver head", "delivered", from, "advertised", head)`

Reply:
This protection is not needed in the end. The test case was a bit faulty and would have passed if we waited out the default 2 minute timeout of the receipt downloads.